From c0683250de3246634c04c2bb412e751e8b89a297 Mon Sep 17 00:00:00 2001
From: Vladimir Panteleev <git@thecybershadow.net>
Date: Tue, 19 Apr 2016 02:28:07 +0000
Subject: [PATCH 3/3] Fix grep dependency on locale for DDoc tests

By default, grep attempts to detect whether an input file is binary,
and if so, in the case of a match it simply prints "Binary file foo
matches". This detection depends on the current locale, as specified
in the environment.

The ddoc12 test contains Unicode characters. When the locale in the
environment is set to en_US.UTF8, this test passes. However, if the
enviroment variables are unset, the test fails, as grep considers
the file as binary. This causes the diff in ddocAny-postscript.sh
to fail, as it will compare the expected result with the string
"Binary file test_results/compilable/ddoc12.html matches".

This test failure was discovered with Digger, which uses a pristine
environment during its builds.
---
 test/compilable/extra-files/ddocAny-postscript.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/compilable/extra-files/ddocAny-postscript.sh b/test/compilable/extra-files/ddocAny-postscript.sh
index 79dcf7af4..01c901bb3 100755
--- a/test/compilable/extra-files/ddocAny-postscript.sh
+++ b/test/compilable/extra-files/ddocAny-postscript.sh
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-grep -v "Generated by Ddoc from" ${RESULTS_DIR}/compilable/ddoc$1.html > ${RESULTS_DIR}/compilable/ddoc$1.html.2
+grep --text -v "Generated by Ddoc from" ${RESULTS_DIR}/compilable/ddoc$1.html > ${RESULTS_DIR}/compilable/ddoc$1.html.2
 diff --strip-trailing-cr compilable/extra-files/ddoc$1.html ${RESULTS_DIR}/compilable/ddoc$1.html.2
 if [ $? -ne 0 ]; then
     exit 1;
-- 
2.11.1

